src/plugins/vis_default_editor/public/components/options/percentage_mode.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiCode, EuiFieldText, EuiFormRow, EuiLink } from '@elastic/eui';
+import { SwitchOption } from './switch';
+import { useKibana } from '../../../../kibana_react/public';
+import { UI_SETTINGS } from '../../../../data/public';
+
+export interface PercentageModeOptionProps {
+  setValue: (
```

Comment:
@sulemanof can we improve that somehow?

Reply:
To protect param values for each param I would suggest the next type
```suggestion
interface Values {
  percentageMode: boolean;
  percentageFormatPattern?: string;
}

  setValue: <ParamaName extends keyof Values>(
    paramName: ParamaName,
    value: Values[ParamaName]
  ) => void;
```

Reply:
It's not suit there because we pass different setValue functions in this component which have in common only these paramNames. If we use this approach we get typescript errors in places where we pass setValues functions. Another options is create a separate functions as setValue and pass them into PercentageModeOption in components where we use our component. But I think it's not good solution because in each place where we use our component we should create a function. I think in this case the best option is leave it as it is now.